### PR TITLE
Unfuck the statue code.

### PR DESCRIPTION
**Pull request by phil235**

### CODE REVIEW

File: code/game/objects/structures/crates_lockers/closets/statue.dm [hunk 1]
```diff
@@ -1,146 +1,106 @@
-/obj/structure/closet/statue
```
Comment (phil235) [inline code/game/objects/structures/crates_lockers/closets/statue.dm]: this is cancer

File: code/game/objects/structures/crates_lockers/closets/statue.dm [hunk 2]
```diff
@@ -1,146 +1,106 @@
...
-		qdel(src)
-		return
-
-	START_PROCESSING(SSobj, src)
+		START_PROCESSING(SSobj, src)
 	..()
-	icon = L.icon
-	icon_state = L.icon_state
-	overlays = L.overlays
-	color = list(rgb(77,77,77), rgb(150,150,150), rgb(28,28,28), rgb(0,0,0))
```
Comment (phil235) [inline code/game/objects/structures/crates_lockers/closets/statue.dm]: Apparently the color change don't seem to work for any mob but humans, no idea why, so I moved the code to human/petrify(). If someone knows how to make the color thing work to make monkeys all white let me know.

File: code/game/objects/structures/crates_lockers/closets/statue.dm [hunk 3]
```diff
@@ -1,146 +1,106 @@
...
-	return
-
-/obj/structure/closet/statue/attack_hand()
-	return
-
-/obj/structure/closet/statue/verb_toggleopen()
-	return
-
-/obj/structure/closet/statue/update_icon()
-	return
```
Comment (phil235) [inline code/game/objects/structures/crates_lockers/closets/statue.dm]: Here's your clue that this thing should never have been a closet in the first place.

File: code/game/objects/structures/statues.dm [hunk 4]
```diff
@@ -89,24 +86,12 @@
...
-				var/ore = text2path("/obj/item/stack/sheet/mineral/[mineralType]")
-				for(var/i = 1, i <= oreAmount, i++)
-					new ore(get_turf(src))
-		else
-			if (mineralType == "metal")
-				var/ore = /obj/item/stack/sheet/metal
-				for(var/i = 3, i <= oreAmount, i++)
-					new ore(get_turf(src))
-			else
-				var/ore = text2path("/obj/item/stack/sheet/mineral/[mineralType]")
```
Comment (phil235) [inline code/game/objects/structures/statues.dm]: nngghnngnggnghghg

File: code/modules/projectiles/projectile/magic.dm [hunk 5]
```diff
@@ -332,35 +332,36 @@
 	damage_type = BURN
 	nodamage = 1
 
-/obj/item/projectile/magic/animate/Bump(atom/change)
```
Comment (phil235) [inline code/modules/projectiles/projectile/magic.dm]: why on Bump(), WHYYYYY

File: code/game/objects/structures/crates_lockers/closets/statue.dm [hunk 6]
```diff
@@ -1,146 +1,106 @@
...
 			L.buckled.unbuckle_mob(L,force=1)
-		L.reset_perspective(src)
-		L.loc = src
-		L.disabilities += MUTE
-		L.faction += "mimic" //Stops mimics from instaqdeling people in statues
 		L.visible_message("<span class='warning'>[L]'s skin rapidly turns to marble!</span>", "<span class='userdanger'>Your body freezes up! Can't... move... can't...  think...</span>")
-
+		L.forceMove(src)
+		L.disabilities |= MUTE
+		L.faction += "mimic" //Stops mimics from instaqdeling people in statues
```
Comment (ChangelingRain) [inline code/game/objects/structures/crates_lockers/closets/statue.dm]: This shouldn't be necessary given that simple animals can't attack statues by default(they could attack closets, though)

**Comment by nullbear** [comment]
Wtf this is so dumb

File: code/game/objects/structures/crates_lockers/closets/statue.dm (hunk 3 above)
Comment (ChangelingRain) [inline code/game/objects/structures/crates_lockers/closets/statue.dm]: I see a lot of clues.

File: code/game/objects/structures/crates_lockers/closets/statue.dm [hunk 7]
```diff
@@ -1,146 +1,106 @@
...
+	if(!petrified_mob)
+		STOP_PROCESSING(SSobj, src)
 	timer--
-	for(var/mob/living/M in src) //Go-go gadget stasis field
-		M.setToxLoss(intialTox)
-		M.adjustFireLoss(intialFire - M.getFireLoss())
-		M.adjustBruteLoss(intialBrute - M.getBruteLoss())
-		M.setOxyLoss(intialOxy)
-		M.Stun(1) //So they can't do anything while petrified
+	petrified_mob.Stun(2) //So they can't do anything while petrified
```
Comment (Durkel) [inline code/game/objects/structures/crates_lockers/closets/statue.dm]: It's a shame the hulk cancer needs to be worked with at all instances, remove hulk when.

File: code/game/objects/structures/statues.dm (hunk 4 above)
Comment (ChangelingRain) [inline code/game/objects/structures/statues.dm]: test2path is ffffucking awful

File: code/game/objects/structures/crates_lockers/closets/statue.dm (hunk 7 above)
Comment (optimumtact) [inline code/game/objects/structures/crates_lockers/closets/statue.dm]: set nomove or notransform

we really need a new flag NOCONTROL that removes all player interaction (click/verb moves)

**Comment by Supermichael777** [comment]
Statues should not have been merged while they were closets

Still cancer but now its not ass cancer

**Pull request merged**